spring-cloud-stream-reactive/src/test/java/org/springframework/cloud/stream/reactive/StreamListenerReactorTests.java

```diff
@@ -165,7 +334,17 @@ public void receive(@Input(Processor.INPUT) Flux<String> input, @Output(Processo
 
 	@EnableBinding(Processor.class)
 	@EnableAutoConfiguration
-	public static class TestInputOutputArgsWithMessage {
+	public static class TestInputOutputArgs2 {
+
+		@StreamListener(Processor.INPUT)
```

Comment:
Can you clarify your comment a bit?

>  I think it would be a convenient for the users to specify the binding targets in whatever places that it actually matters.

I specifically disagree with:

>For instance, user knows either @StreamListener or the method parameter with @Input accepts the inbound target and they can specify wherever.

I am specifically concerned about cases where inputs and outputs have the same type, e.g. `MessageChannel` - so it is impossible to tell whether the parameter is an input or an output - ordering shouldn't be a criterion. 

To me, it would be much clearer to the user if we supported only this:

a) `@StreamListener`/`@SendTo` without any `@Input`/`@Output` annotations wherever there's a single input/single or no output use case. This would align individual messages and bindable types.

b) `@StreamListener` + `@Input`/`@Output` to cover all the cases where we use bindable types. We could use `SendTo` as an alias for `@Output` in that case. 

I don't see any feature loss, but a lot fewer corner cases and idiomatic subtleties for us to deal with in the future.